SAKPaaS/src/app/generated/api-configuration.ts

```diff
@@ -8,7 +8,7 @@ import { Injectable } from '@angular/core';
   providedIn: 'root',
 })
 export class ApiConfiguration {
-  rootUrl: string = 'https://virtserver.swaggerhub.com/SAKPaaS/SAKPaaS/1.0.0';
+  rootUrl: string;
```

Comment:
Ehm wenn du das hier änderst, wird das beim Neu-Generieren der API überschrieben… Perspektivisch müsste das also irgendwie über einen Provider in das ApiModule injected werden… I guess.

Reply:
@timgrohmann Ähm ja hast Recht, aber der Standard wird eh beim Initialisieren der Map überschrieben. In der Loader Function:
```
export function load(http: HttpClient, config: ApiConfiguration): (() => Promise<boolean>) {
  return (): Promise<boolean> => {
    return new Promise<boolean>((resolve: (a: boolean) => void): void => {
       http.get('./config/config.json')
         .pipe(
           map((x: any) => {
             config.rootUrl = x.rootUrl + '/v1';       // HIER BEIM DEPLOYEN
             resolve(true);
           }),
           catchError((x: { status: number }, caught: Observable<void>): ObservableInput<{}> => {
             if (x.status !== 404) {
               resolve(false);
             }
             config.rootUrl = 'https://api.happyhamster.org/v1';     // HIER BEIM ENTWICKELN
             resolve(true);
             return of({});
           })
         ).subscribe();
    });
  };
```

Reply:
Ja okay, good point. Dann würde ich vllt in der `openapi.yml` den aktuellen Server irgendwie angeben (also nicht _swaggerhub_)

Reply:
öhm aber dann muss ich die yaml bei jeden DB-Update anpassen oder nicht?

Reply:
Ja gut aber so müsstest du den Default-Wert in dem Loader jedes mal anpassen. Also irgendwo muss es ja hin, oder? @nclskfm

Reply:
Hmm also sonst stand es halt im Import vom ApiModule in app.module.ts, auch nicht in der yaml.

Also ich finde es besser im Loader (Vielleicht die Funktion umbenennen, ApiLoader zb?) als in einer yaml, die jedes Mal umgeschrieben werden muss, wenn wir das aktualisieren. :D @timgrohmann